storage/backup/backup-service/src/handlers/utils.rs

```diff
@@ -93,12 +178,17 @@ where
     I: Iterator<Item = Result<R>>,
     R: Serialize,
 {
-    send_size_prefixed_bcs_bytes_impl(iter_res, &mut sender)
-        .await
-        .unwrap_or_else(|e| {
-            warn!("Failed writing to output http body: {:?}", e);
-            sender.abort()
-        });
+    match send_size_prefixed_bcs_bytes_impl(iter_res, &mut sender).await {
+        Ok(()) => {
+            if let Err(e) = sender.finish().await {
+                warn!("Failed to finish http body: {:?}", e);
+            }
+        },
+        Err(e) => {
+            warn!("Failed writing http body: {:?}", e);
+            sender.abort().await;
```

Comment:
if error happens, we can still use `finish`? any reason for introducing one more abort function?